external/zsh/_msfvenom

```diff
@@ -207,6 +207,19 @@ _msfvenom_platform() {
   _describe -t platforms 'available platforms' _msfvenom_platforms_list || compadd "$@"
 }
 
+_msfvenom_payload() {
+  local cacheFile="$HOME/.msf4/store/modules_metadata.json"
+  local -a _msfvenom_payloads_list
+  [ -f "$cacheFile" ] || cacheFile="/opt/metasploit/db/modules_metadata_base.json"
+  if [ ! -f "$cacheFile" ]; then
+    _message -r "Cannot find metasploit cache file. Run msfconsole to populate it"
+    compadd "$@"
+  else
+    _msfvenom_payloads_list=("${(f)$(sed -n '/"type": "payload"/,/"ref_name"/p' "$cacheFile" | grep -E '(ref_name|description)' | cut -d '"' -f 4 | sed -n 'h;n;p;g;p' | sed 'N;s/\n/:/; s/\\n.*$//')}")
```

Comment:
@jmartin-r7 @acammack-r7 Just pinging yourselves to confirm if you'd be happy with this new approach which makes use of the modules metadata json file for zsh completion? :+1:

Reply:
I like this approach as a compromise to weekly cache.

I believe it might be reasonable to update this to an inline ruby script to us the built in parser on the json.  Happy to defer as a future improvement.